.github/workflows/configs-docs-scraping.yml

```diff
@@ -0,0 +1,166 @@
+
+          undocumented_next_configs=$(echo "$output" | grep -E "__gh_output__undocumented_next_configs=")
+          undocumented_next_configs_value="${undocumented_next_configs#*=}"
+          echo "undocumented_next_configs=$undocumented_next_configs_value" >> $GITHUB_OUTPUT
+
+          documented_non_next_configs=$(echo "$output" | grep -E "__gh_output__documented_non_next_configs=")
+          documented_non_next_configs_value="${documented_non_next_configs#*=}"
+          echo "documented_non_next_configs=$documented_non_next_configs_value" >> $GITHUB_OUTPUT
+      - name: Check result
+        run: |
+          if [[ "${{ steps.run-checker.outputs.result }}" = "no-next-configs-detected" ]]
+          then
+            echo "ERROR! No next configs were detected, the next docs might have changed!"
+            exit 1
+          elif [[ "${{ steps.run-checker.outputs.result }}" = "no-next-configs-detected" ]]
+          then
+            echo "ERROR! No next-on-pages configs were detected, the docs might have changed!"
+            exit 1
+          elif [[ "${{ steps.run-checker.outputs.result }}" = "no-next-configs-detected" ]]
+          then
+            echo "The next-on-pages documentation is up to date"
+          else
+            echo "The next-on-pages documentation is out of date"
+          fi
+
+  generate-configs-table:
+    name: Generate configs table
+    needs: check_docs
+    if: needs.check_docs.outputs.result == 'out-of-date'
+    runs-on: ubuntu-latest
+    outputs:
+      table: ${{ steps.run-checker.outputs.table }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Node.js 16.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Checker
+        id: run-checker
+        working-directory: internal-packages/docs-scraper
+        run: |
+          output=$(GH_ACTION=true npm run generate-configs-table)
+
+          table=$(echo "$output" | grep -E '^\|.*' )
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "table<<$EOF" >> $GITHUB_OUTPUT
+          echo "$table" >> $GITHUB_OUTPUT
+          echo "$EOF" >> $GITHUB_OUTPUT
+
+  create_or_update_issue:
+    name: Create or Update Issue
+    needs: [check_docs, generate-configs-table]
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: Update Issue
+        uses: actions/github-script@v6
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          undocumentedNextConfigs: ${{ needs.check_docs.outputs.undocumented_next_configs }}
+          documentedNonNextConfigs: ${{ needs.check_docs.outputs.documented_non_next_configs }}
+          configsTable: ${{ needs.generate-configs-table.outputs.table }}
+        with:
+          script: |
+            const label = 'outdated-config-docs';
```

Comment:
Is it possible to make this an actual script in the repo that the workflow just invokes instead of inlining it here like this?

Reply:
ah, yeah awesome suggestion!

I used this as the action gives you these `github` and `context` globally available variables, but yeah this does look quite heavy handed and probably not the best way to do it

I think that I can take those `github` and `context` object from this: https://github.com/actions/toolkit/tree/main/packages/github
so maybe I can just install the package and call a script from here

I'll give that a shot! thanks a lot 😄

Reply:
Updated :smiley: